Packages/com.llealloo.audiolink/Runtime/Scripts/YtdlpPlayer.cs

```diff
@@ -0,0 +1,303 @@
+            ytdlProcess.StartInfo.FileName = _ytdlpPath;
+            ytdlProcess.StartInfo.Arguments = $"--no-check-certificate --no-cache-dir --rm-cache-dir -f \"mp4[height<=?{resolution}]/best[height<=?{resolution}]\" --get-url \"{url}\"";
+
+            Debug.Log($"[AudioLink] Attempting to resolve URL '{url}'");
+
+            try
+            {
+                ytdlProcess.Start();
+                _runningYtdlProcesses.Add(ytdlProcess);
+
+                while (!ytdlProcess.HasExited)
+                    new WaitForSeconds(0.1f);
+
+                _runningYtdlProcesses.Remove(ytdlProcess);
+
+                return ytdlProcess.StandardOutput.ReadLine();
+            }
+            catch(Exception e)
+            {
+                Debug.LogWarning($"[AudioLink] Unable to resolve URL '{url}' : " + e.Message);
+                return null;
+            }
+        }
+
+        /// <summary> Resolves a URL and set a videoplayer's URL to the resolved URL. </summary>
+        /// <param name="url">URL to resolve for playback</param>
+        /// <param name="resolution">Resolution (vertical) to request from yt-dlp</param>
+        /// <param name="player">VideoPlayer component to set URL on</param>
+        public static void ResolveAndSet(string url, int resolution, VideoPlayer player)
+        {
+            try
+            {
+                string resolved = Resolve(url, resolution);
+                if(resolved != null)
+                    player.url = resolved;
+            }
+            catch(Exception e)
+            {
+                Debug.LogWarning($"[AudioLink] Unable to play URL '{url}' : " + e.Message);
+            }
+        }
+
+        /// <summary> Cleans up any remaining YTDL processes from this play, in case they don't clean up after themselves. </summary>
+        private static void PlayModeChanged(PlayModeStateChange change)
+        {
+            if (change == PlayModeStateChange.ExitingPlayMode)
+            {
+                foreach (Process process in _runningYtdlProcesses)
+                {
+                    if (!process.HasExited)
+                    {
+                        process.Close();
+                    }
+                }
+
+                _runningYtdlProcesses.Clear();
+            }
+        }
+
+        /// <summary> Download yt-dlp to the AudioLink folder. </summary>
+        private static void DownloadYtdlp()
+        {
+            WebClient webClient = new WebClient();
+            try
+            {
+                webClient.DownloadFile(new Uri(_ytdlpDownloadURL), _localYtdlpPath);
+                Debug.Log($"[AudioLink] yt-dlp downloaded to '{_ytdlpPath}'");
+                AssetDatabase.Refresh();
+            }
+            catch(Exception e)
+            {
+                Debug.LogWarning($"[AudioLink] Failed to download yt-dlp from '{_ytdlpDownloadURL}' : " + e.Message);
+            }
+            webClient.Dispose();
```

Comment:
Use a `using` statement instead of a manual dispose

Reply:
Resolved in 631f888e53bf119a63570ed455611d1e38ef151d